stable/prometheus-to-sd/templates/deployment.yaml

```diff
@@ -0,0 +1,36 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ .Release.Name }}
```

Comment:
Why did you change this? Should be `{{ template "prometheus-to-sd.fullname" . }}`

Reply:
`{{ template "prometheus-to-sd.fullname" . }}` causes ugly resource names. The problem is discussed in https://github.com/kubernetes/helm/issues/2062  and I think it might be solved in Helm 3 if  https://github.com/kubernetes/helm/issues/2060 is approved. `{{ .Release.Name }}` is unique and can be safely used as the name of the resource.

Reply:
I had to cherry-pick some commits related to circle CI because the PR was opened long before test changes applied, so the tests were failing.

Reply:
Currently, the convention is to use `{{ template "prometheus-to-sd.fullname" . }}`. This is what `helm create` gives you. Chart examples use this as well. Please follow this best practice. You may adjust the implementation of the template providing a fullnameOverride if you like. Compare https://github.com/kubernetes/charts/blob/master/incubator/common/templates/_fullname.tpl#L20-L29.

Reply:
Here's another alternative that should work for you.
https://github.com/compleatang/charts/blob/7cfe269290af6c2cc1e0ae9e801ad31a9099ba8d/incubator/oauth-proxy/templates/_helpers.tpl#L13-L19